controlpanel/api/views/tools.py

```diff
@@ -36,4 +36,9 @@ def create(self, request):
         return Response({}, status=status.HTTP_201_CREATED)
 
     def get_queryset(self):
-        return ToolDeployment.objects.filter(user=self.request.user)
+        user = self.request.user
+        id_token = user.get_id_token()
+        return ToolDeployment.objects.filter(
+            user=user,
+            id_token=id_token,
+        )
```

Comment:
This is a bit weird, because it reads like you're filtering based on an `id_token` field. I suppose that shows the problem with mimicking a Django model.

Reply:
Thanks for the review @andyhd :)

That may be a next step in the refactoring. I haven't looked at the consequences of removing the model altogether, but I imagine there is something more done in the views/serialisers to make it look like a real model so I'm not touching at the moment.

I could still group these functions in a `cluster.ToolDeployment` class and use it from the `models.ToolDeployment` class as a stepping stone to future changes but I'll probably leave as a separate PR anyway as this is enough for a change set IMHO.

Another thing I'm wrapping my head around is the helm install subprocess polling which is currently distributed between `cluster`module (`deploy_tool()`), `models.ToolDeployment` (`get_status()`/`_poll()`) and `consumers` module (`wait_for_deployment()`). We could probably move this "waiting/checking" logic into the class instance but I didn't think too much about it yet.